src/klr.net45/klr.net45.cpp

```diff
@@ -8,20 +8,25 @@
 
 IKatanaManagerPtr g_katanaManager;
 
-
 extern "C" __declspec(dllexport) HRESULT __stdcall CallApplicationMain(PCALL_APPLICATION_MAIN_DATA data)
 {
     HRESULT hr = S_OK;
-    
+
     IKatanaManagerPtr manager = new ComObject<KatanaManager, IKatanaManager>();
 
     g_katanaManager = manager;
-    _HR(manager->InitializeRuntime(data->applicationBase));
-    g_katanaManager = NULL;
-
-    _HR(manager->CallApplicationMain(data->argc, data->argv));
 
-    data->exitcode = hr;
+    hr = manager->InitializeRuntime(data->applicationBase);
+    if (SUCCEEDED(hr))
+    {
+        g_katanaManager = NULL;
+        data->exitcode = manager->CallApplicationMain(data->argc, data->argv);
```

Comment:
The return value here is not an actually `HRESULT`, but a `int` returns from the application. It can't be judge by `SUCCEEDED(hr)`. No message should be print no matter what the result is. Output should be controlled by the application itself.